tests/conftest.py

```diff
@@ -10,3 +14,43 @@ def pytest_addoption(parser):
         type=pathlib.Path,
         help="Root of the ISO file tree."
     )
+
+# Fixtures {{{
+
+
+@pytest.fixture(scope="module")
+def version(request, host):
+    iso_root = request.config.getoption("--iso-root")
+    product_path = iso_root / "product.txt"
+    with host.sudo():
+        return host.check_output(
+            'source %s && echo $SHORT_VERSION', str(product_path)
+        )
+
+
+@pytest.fixture(scope="module")
+def kubeconfig_data(request, host):
+    """Fixture to generate a kubeconfig file for remote usage."""
+    with host.sudo():
+        kubeconfig_file = host.file('/etc/kubernetes/admin.conf')
+        if not kubeconfig_file.exists:
+            pytest.skip(
+                "Must be run on bootstrap node, or have an existing file at "
+                "/etc/kubernetes/admin.conf"
+            )
+        return yaml.safe_load(kubeconfig_file.content_string)
+
+
+@pytest.fixture
+def kubeconfig(kubeconfig_data, tmp_path):
+    kubeconfig_path = tmp_path / "admin.conf"
+    kubeconfig_path.write_text(yaml.dump(kubeconfig_data), encoding='utf-8')
+    return str(kubeconfig_path)  # Need Python 3.6 to open() a Path object
+
+
+@pytest.fixture
+def k8s_client(kubeconfig):
+    kubernetes.config.load_kube_config(config_file=kubeconfig)
```

Comment:
Sure, will do!

Reply:
So I still have a `k8s_client` for clients exposing a single API, and `k8s_apiclient` that returns an `ApiClient` to use in `k8s_client`. By default, we use `CoreV1Api`, but it can be parametrized with `pytest.mark.parametrize`.